src/applications/facility-locator/components/ResultsList.jsx

```diff
@@ -37,6 +44,66 @@ class ResultsList extends Component {
     }
   }
 
+  /**
+   * Returns Result items by type
+   * @param isMobile boolean
+   * @param query object
+   * @param results array list
+   * @returns [] list of results
+   */
+  renderResultItems(isMobile, query, results) {
+    return results.map(r => {
+      let item;
+      switch (query.facilityType) {
+        case 'health':
+        case 'cemetery':
+        case 'benefits':
+        case 'vet_center':
+          item = <VaFacilityResult location={r} query={query} />;
+          break;
+        case 'provider':
+          // Support non va urgent care search through ccp option
+          if (query.serviceType === CLINIC_URGENTCARE_SERVICE) {
+            item = <UrgentCareResult provider={r} query={query} />;
+          } else {
+            item = <CCProviderResult provider={r} query={query} />;
+          }
+          break;
+        case 'pharmacy':
+          item = <PharmacyResult provider={r} query={query} />;
+          break;
+        case 'urgent_care':
+          if (query.serviceType === 'NonVAUrgentCare') {
+            item = <UrgentCareResult provider={r} query={query} />;
+          } else if (query.serviceType === 'UrgentCare') {
+            item = <VaFacilityResult location={r} query={query} />;
+          } else if (
+            !query.serviceType ||
+            query.serviceType === 'AllUrgentCare'
+          ) {
+            // New option
+            // All mash up
+            if (r.type === LocationType.CC_PROVIDER) {
+              item = <UrgentCareResult provider={r} query={query} />;
+            } else {
+              item = <VaFacilityResult location={r} query={query} />;
+            }
```

Comment:
What do you think about simplifying this logic like so:

```suggestion
            if (r.type === LocationType.CC_PROVIDER) {
              item = <UrgentCareResult provider={r} query={query} />;
            } else {
              item = <VaFacilityResult location={r} query={query} />;
            }
```